graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -570,8 +573,8 @@ def _validate_fold_has_outputs_or_count_filter(fold_scope_location, fold_has_cou
 
     # At least one output in the outputs list must point to the fold_scope_location,
     # or the scope corresponding to fold_scope_location had no @outputs and is illegal.
-    for output in six.itervalues(outputs):
-        if output['fold'] == fold_scope_location:
+    for _, output_info in metadata.outputs:
+        if output_info.fold == fold_scope_location:
```

Comment:
Let's try to get rid of the `.fold` attribute, since it was a hack meant to work around the lack of a metadata table. I suspect you'll be able to reconstruct its value from the `.location` attribute that is available on the output info object.